Improve error message for Git LFS related failures

As part of #4421, we moved all parquet files from the core repo to Git LFS. So now, on pulling the core repo, these files don't get fetched by default in case Git LFS is not installed on the system. This can lead to errors like the following while running tests locally:
```
    Caused by: io.deephaven.parquet.base.ParquetFileReaderException: extensions/parquet/table/build/resources/test/e1/uncompressed.parquet is not a Parquet file. expected magic number at tail [80, 65, 82, 49] but found [55, 49, 48, 10]
    at app//io.deephaven.parquet.base.ParquetFileReader.<init>(ParquetFileReader.java:62)
    at app//io.deephaven.parquet.table.ParquetTools.getParquetFileReaderChecked(ParquetTools.java:801)
    at app//io.deephaven.parquet.table.ParquetTools.getParquetFileReader(ParquetTools.java:784)
    ... 50 more
```
The fix is to run `git lfs install; git lfs pull` inside the repo.
We should make the error message more clear with the above instructions to fix it. Also, we should update the README to mention setup instructions for Git LFS.

--- a/README.md
+++ b/README.md
@@ -60,7 +60,7 @@ docker run hello-world
 If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/quickstart#prerequisites) for installation instructions.
 
 
-For running unit tests, you will also need Git LFS. Run the following commands inside the core repo to fetch all required files from LFS:
+For running the unit tests, you will also need to install [Git LFS](https://git-lfs.com/) and fetch all the required files. Run the following commands from inside the directory:
 ```
 git lfs install
 git lfs pull

--- a/README.md
+++ b/README.md
@@ -60,6 +60,12 @@ docker run hello-world
 If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/quickstart#prerequisites) for installation instructions.
 
 
+For running unit tests, you will also need Git LFS. Run the following commands inside the core repo to fetch all required files from LFS:
+```
+git lfs install
+git lfs pull
+```
+
 ### Create deployment
 
 A directory must be created to store files and mount points for your deployment.  Here, we are using the `deephaven-deployment` directory.  

--- a/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/InvalidParquetFileException.java
+++ b/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/InvalidParquetFileException.java
@@ -1,5 +1,8 @@
 package io.deephaven.parquet.base;
 
+/**
+ * Exception thrown when trying to read an invalid Parquet file.
+ */
 public class InvalidParquetFileException extends ParquetFileReaderException {
     InvalidParquetFileException(String message) {
         super(message);

--- a/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/InvalidParquetFileException.java
+++ b/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/InvalidParquetFileException.java
@@ -1,0 +1,7 @@
+package io.deephaven.parquet.base;
+
+public class InvalidParquetFileException extends ParquetFileReaderException {
+    InvalidParquetFileException(String message) {
+        super(message);
+    }
+}

--- a/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java
+++ b/extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java
@@ -46,7 +46,7 @@ public class ParquetFileReader {
             final long fileLen = readChannel.size();
             if (fileLen < MAGIC.length + FOOTER_LENGTH_SIZE + MAGIC.length) { // MAGIC + data + footer +
                 // footerIndex + MAGIC
-                throw new ParquetFileReaderException(
+                throw new InvalidParquetFileException(
                         filePath + " is not a Parquet file (too small length: " + fileLen + ")");
             }
 
@@ -57,13 +57,13 @@ public class ParquetFileReader {
             final byte[] magic = new byte[MAGIC.length];
             Helpers.readBytes(readChannel, magic);
             if (!Arrays.equals(MAGIC, magic)) {
-                throw new ParquetFileReaderException(
+                throw new InvalidParquetFileException(
                         filePath + " is not a Parquet file. expected magic number at tail "
                                 + Arrays.toString(MAGIC) + " but found " + Arrays.toString(magic));
             }
             final long footerIndex = footerLengthIndex - footerLength;
             if (footerIndex < MAGIC.length || footerIndex >= footerLengthIndex) {
-                throw new ParquetFileReaderException(
+                throw new InvalidParquetFileException(
                         "corrupted file: the footer index is not within the file: " + footerIndex);
             }
             readChannel.position(footerIndex);

--- a/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java
+++ b/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java
@@ -24,6 +24,7 @@ import io.deephaven.engine.testutil.TstUtils;
 import io.deephaven.engine.testutil.junit4.EngineCleanup;
 import io.deephaven.engine.util.BigDecimalUtils;
 import io.deephaven.engine.util.file.TrackedFileHandleFactory;
+import io.deephaven.parquet.base.InvalidParquetFileException;
 import io.deephaven.parquet.table.location.ParquetTableLocationKey;
 import io.deephaven.stringset.ArrayStringSet;
 import io.deephaven.engine.table.Table;
@@ -1038,8 +1039,18 @@ public class ParquetTableReadWriteTest {
     public void verifyPyArrowStatistics() {
         final String path = ParquetTableReadWriteTest.class.getResource("/e0/pyarrow_stats.parquet").getFile();
         final File pyarrowDest = new File(path);
-        final Table pyarrowFromDisk = ParquetTools.readTable(pyarrowDest);
-
+        final Table pyarrowFromDisk;
+        try {
+            pyarrowFromDisk = ParquetTools.readTable(pyarrowDest);
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof InvalidParquetFileException) {
+                final String InvalidParquetFileErrorMsgString = "Invalid parquet file detected, please ensure the " +
+                        "file is fetched properly from Git LFS. Run commands 'git lfs install; git lfs pull' inside " +
+                        "the repo to pull the files from LFS. Check cause of exception for more details.";
+                throw new UncheckedDeephavenException(InvalidParquetFileErrorMsgString, e.getCause());
+            }
+            throw e;
+        }
         // Verify that our verification code works for a pyarrow generated table.
         assertTableStatistics(pyarrowFromDisk, pyarrowDest);
 

--- a/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java
+++ b/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java
@@ -423,10 +423,9 @@ public class TestParquetTools {
                 t -> t.updateView("Y = Z", "Y = X").where("Y % 2 == 0"));
     }
 
-    private static final String invalidParquetFileErrorMsgString = "Invalid parquet file detected, please ensure these "
-            +
-            "files are fetched properly from Git LFS. You can run commands 'git lfs install; git lfs pull' to fetch" +
-            " the files. Check cause of exception for more details.";
+    private static final String InvalidParquetFileErrorMsgString = "Invalid parquet file detected, please ensure " +
+            "the file is fetched properly from Git LFS. Run commands 'git lfs install; git lfs pull' inside the repo " +
+            "to pull the files from LFS. Check cause of exception for more details.";
 
     @Test
     public void e0() {
@@ -448,7 +447,7 @@ public class TestParquetTools {
             assertTableEquals(uncompressed, zstd);
         } catch (RuntimeException e) {
             if (e.getCause() instanceof InvalidParquetFileException) {
-                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+                throw new UncheckedDeephavenException(InvalidParquetFileErrorMsgString, e.getCause());
             }
         }
     }
@@ -473,7 +472,7 @@ public class TestParquetTools {
             assertTableEquals(uncompressed, zstd);
         } catch (RuntimeException e) {
             if (e.getCause() instanceof InvalidParquetFileException) {
-                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+                throw new UncheckedDeephavenException(InvalidParquetFileErrorMsgString, e.getCause());
             }
         }
     }
@@ -498,7 +497,7 @@ public class TestParquetTools {
             assertTableEquals(uncompressed, zstd);
         } catch (RuntimeException e) {
             if (e.getCause() instanceof InvalidParquetFileException) {
-                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+                throw new UncheckedDeephavenException(InvalidParquetFileErrorMsgString, e.getCause());
             }
         }
     }

--- a/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java
+++ b/extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java
@@ -14,6 +14,7 @@ import io.deephaven.engine.table.impl.InMemoryTable;
 import io.deephaven.engine.table.impl.locations.TableDataException;
 import io.deephaven.engine.testutil.junit4.EngineCleanup;
 import io.deephaven.engine.util.TableTools;
+import io.deephaven.parquet.base.InvalidParquetFileException;
 import io.deephaven.parquet.table.layout.ParquetKeyValuePartitionedLayout;
 import io.deephaven.stringset.HashStringSet;
 import io.deephaven.stringset.StringSet;
@@ -422,58 +423,84 @@ public class TestParquetTools {
                 t -> t.updateView("Y = Z", "Y = X").where("Y % 2 == 0"));
     }
 
+    private static final String invalidParquetFileErrorMsgString = "Invalid parquet file detected, please ensure these "
+            +
+            "files are fetched properly from Git LFS. You can run commands 'git lfs install; git lfs pull' to fetch" +
+            " the files. Check cause of exception for more details.";
+
     @Test
     public void e0() {
-        final Table uncompressed =
-                ParquetTools.readTable(TestParquetTools.class.getResource("/e0/uncompressed.parquet").getFile());
+        try {
+            final Table uncompressed =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e0/uncompressed.parquet").getFile());
 
-        final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/gzip.parquet").getFile());
-        assertTableEquals(uncompressed, gzip);
+            final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/gzip.parquet").getFile());
+            assertTableEquals(uncompressed, gzip);
 
-        final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/lz4.parquet").getFile());
-        assertTableEquals(uncompressed, lz4);
+            final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/lz4.parquet").getFile());
+            assertTableEquals(uncompressed, lz4);
 
-        final Table snappy = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/snappy.parquet").getFile());
-        assertTableEquals(uncompressed, snappy);
+            final Table snappy =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e0/snappy.parquet").getFile());
+            assertTableEquals(uncompressed, snappy);
 
-        final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/zstd.parquet").getFile());
-        assertTableEquals(uncompressed, zstd);
+            final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e0/zstd.parquet").getFile());
+            assertTableEquals(uncompressed, zstd);
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof InvalidParquetFileException) {
+                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+            }
+        }
     }
 
     @Test
     public void e1() {
-        final Table uncompressed =
-                ParquetTools.readTable(TestParquetTools.class.getResource("/e1/uncompressed.parquet").getFile());
+        try {
+            final Table uncompressed =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e1/uncompressed.parquet").getFile());
 
-        final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/gzip.parquet").getFile());
-        assertTableEquals(uncompressed, gzip);
+            final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/gzip.parquet").getFile());
+            assertTableEquals(uncompressed, gzip);
 
-        final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/lz4.parquet").getFile());
-        assertTableEquals(uncompressed, lz4);
+            final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/lz4.parquet").getFile());
+            assertTableEquals(uncompressed, lz4);
 
-        final Table snappy = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/snappy.parquet").getFile());
-        assertTableEquals(uncompressed, snappy);
+            final Table snappy =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e1/snappy.parquet").getFile());
+            assertTableEquals(uncompressed, snappy);
 
-        final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/zstd.parquet").getFile());
-        assertTableEquals(uncompressed, zstd);
+            final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e1/zstd.parquet").getFile());
+            assertTableEquals(uncompressed, zstd);
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof InvalidParquetFileException) {
+                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+            }
+        }
     }
 
     @Test
     public void e2() {
-        final Table uncompressed =
-                ParquetTools.readTable(TestParquetTools.class.getResource("/e2/uncompressed.parquet").getFile());
+        try {
+            final Table uncompressed =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e2/uncompressed.parquet").getFile());
 
-        final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/gzip.parquet").getFile());
-        assertTableEquals(uncompressed, gzip);
+            final Table gzip = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/gzip.parquet").getFile());
+            assertTableEquals(uncompressed, gzip);
 
-        final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/lz4.parquet").getFile());
-        assertTableEquals(uncompressed, lz4);
+            final Table lz4 = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/lz4.parquet").getFile());
+            assertTableEquals(uncompressed, lz4);
 
-        final Table snappy = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/snappy.parquet").getFile());
-        assertTableEquals(uncompressed, snappy);
+            final Table snappy =
+                    ParquetTools.readTable(TestParquetTools.class.getResource("/e2/snappy.parquet").getFile());
+            assertTableEquals(uncompressed, snappy);
 
-        final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/zstd.parquet").getFile());
-        assertTableEquals(uncompressed, zstd);
+            final Table zstd = ParquetTools.readTable(TestParquetTools.class.getResource("/e2/zstd.parquet").getFile());
+            assertTableEquals(uncompressed, zstd);
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof InvalidParquetFileException) {
+                throw new UncheckedDeephavenException(invalidParquetFileErrorMsgString, e.getCause());
+            }
+        }
     }
 
     private void testWriteRead(Table source, Function<Table, Table> transform) {

